app/models/post.rb

```diff
@@ -610,6 +610,18 @@ def url(opts = nil)
     end
   end
 
+  def canonical_url
+    topic_view = TopicView.new(topic, nil, post_number: post_number)
```

Comment:
Just thinking out loud here but I wonder if we should skip this if `post_number == 1` as page will always be 1.

Reply:
Probably a safe micro optimisation, but I wonder if topic_view should do it instead. Can probably wait. 

There is also a redis lookup topic view does that it could delay.